### PR TITLE
Some refactoring and doc fixes

### PR DESCRIPTION
**Pull request by FedorLap2006**
Refactored some of your code, plus changed documentation to satisfy Go's standards.

### CODE REVIEW

**Review by kislball — CHANGES_REQUESTED** [review]
Everything is nice, but there is a moment I don't understand

File: cache.go [hunk 1]
```diff
@@ -65,18 +79,16 @@ func (c *MapCacheProvider) Clear(namespace string) (err error) {
 	ns := c.getNamespace(namespace)
 	ns.Lock()
 	defer ns.Unlock()
-	ns.data = map[string]interface{}{}
+	ns.data = make(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (FedorLap2006) [reply cache.go:82]: It's more clear and less confusing. Also it's quite the standard to initialise an empty map like that.

**Comment by kislball** [comment]
Have you done `go vet` on the project?

**Review by kislball — COMMENTED** [review]
NIce, merging

**Review by kislball — APPROVED** [review]
Nice!

**Pull request merged**